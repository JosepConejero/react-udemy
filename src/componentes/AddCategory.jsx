import { useState } from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ onNewCategory }) => {
  const [inputValue, setInputValue] = useState("");

  const onChangeHandler = ({ target: { value } }) => {
    setInputValue(value);
  };

  const onSubmit = (event) => {
    //console.log("Entra en el onSubmit desde el test");
    event.preventDefault();
    if (inputValue.trim().length <= 1) return;
    //console.log("VALORES DE MÁS DE 1 CARÁCTER");
    onNewCategory(inputValue.trim());
    setInputValue("");
  };

  return (
    <form onSubmit={onSubmit} aria-label="formElement">
      <input
        type="text"
        placeholder="Buscar gifs"
        value={inputValue}
        autoFocus
        onChange={onChangeHandler}
      />
    </form>
  );
};

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
};
