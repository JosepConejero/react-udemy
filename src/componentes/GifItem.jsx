import PropTypes from "prop-types";

export const GifItem = ({ title, url, id, onDelete }) => {
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
      <button onClick={onDelete}>delete</button>
    </div>
  );
};

GifItem.propTypes = {
  title: PropTypes.string,
  url: PropTypes.string.isRequired,
  onDelete: PropTypes.func.isRequired,
};
