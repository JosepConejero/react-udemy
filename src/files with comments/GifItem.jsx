//export const GifItem = (props) => {
//export const GifItem = ({ image }) => {
export const GifItem = ({ title, url, id }) => {
  //console.log(props);
  return (
    <div className="card">
      <img src={url} alt={title} />
      <p>{title}</p>
    </div>
  );
};

//<li key={id}>{title}</li>;
