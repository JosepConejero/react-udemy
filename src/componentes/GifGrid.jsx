import { useFetchGifs } from "../hooks/useFetchGifs";
//import { useEffect, useState } from "react";
import { GifItem } from "./GifItem";
//import { useFetchGifs, GifItem } from "../indexes/varios.js"; // este lo he creado yo
//import { getGifs } from "../helpers/getGifs";

export const GifGrid = ({ category }) => {
  const { images, isLoading, deleteHandler } = useFetchGifs(category);

  /* const [images, setImages] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const getImages = async () => {
    const newImages = await getGifs(category);
    setImages(newImages);
    setIsLoading(false);
  };

  useEffect(() => {
    getImages();
  }, []);

  const deleteHandler = (id) => {
    const newImages = images.filter((image) => image.id !== id);
    setImages(newImages);
  }; */

  return (
    <>
      <h3>{category}</h3>
      {/* {isLoading ? <h2>Cargando...</h2> : null} */}
      {isLoading && <h2>Cargando...</h2>}
      <div className="card-grid">
        {images.map((image) => (
          <GifItem
            key={image.id}
            onDelete={() => deleteHandler(image.id)}
            {...image}
          />
        ))}
      </div>
    </>
  );
};
