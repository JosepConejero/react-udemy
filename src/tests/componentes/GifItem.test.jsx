/* eslint-disable testing-library/no-debugging-utils */
import { render, screen } from "@testing-library/react";
import { GifItem } from "../../componentes/GifItem";

describe("Pruebas sobre el componente GifItem", () => {
  const title = "Redeña, ara sí";
  const url = "https://";
  const onDelete = () => {};

  test("Debe hace match con el snapshot", () => {
    const { container } = render(
      <GifItem title={title} url={url} onDelete={onDelete} />
    );
    expect(container).toMatchSnapshot();
  });

  test("Debe mostrar una imagen con el url y el alt indicados", () => {
    render(<GifItem title={title} url={url} onDelete={onDelete} />);
    //screen.debug();
    //expect(screen.getByText(/redeña/i)).toContain(title);
    //console.log(screen.getByRole("img").alt);
    //expect(screen.getByRole("img").title).toBe(title);
    /*   expect(screen.getByRole("img").src).toBe(url);
    expect(screen.getByRole("img").alt).toBe(title); */
    const { src, alt } = screen.getByRole("img");
    expect(src).toBe(url);
    expect(alt).toBe(title);
  });

  test("Debe mostrar el título en el componente", () => {
    render(<GifItem title={title} url={url} onDelete={onDelete} />);
    const parr = screen.getByText(/redeña/i);
    expect(parr.textContent).toContain(title);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
