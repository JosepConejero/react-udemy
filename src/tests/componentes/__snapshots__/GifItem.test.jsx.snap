// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre el componente GifItem Debe hace match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Redeña, ara sí"
      src="https://"
    />
    <p>
      Redeña, ara sí
    </p>
    <button>
      delete
    </button>
  </div>
</div>
`;
